hydrant/adapters/csv.py

```diff
@@ -31,19 +31,26 @@ class CSV_Parser(object):
     def __init__(self, filepath):
         self.filepath = filepath
         self._headers = None
+        self.csv_dialect = None
 
     @property
     def headers(self):
         if self._headers:
             return self._headers
 
         with open(self.filepath, 'r') as csv_file:
-            reader = csv.DictReader(csv_file)
+            if not self.csv_dialect:
+                self.csv_dialect = csv.Sniffer().sniff(csv_file.read(1024))
+                csv_file.seek(0)
+            reader = csv.DictReader(csv_file, dialect=self.csv_dialect)
             self._headers = [field for field in reader.fieldnames]
         return self._headers
 
     def rows(self):
         with open(self.filepath, 'r') as csv_file:
-            reader = csv.DictReader(csv_file)
+            if not self.csv_dialect:
+                self.csv_dialect = csv.Sniffer().sniff(csv_file.read(1024))
```

Comment:
Might be nice to have a log message when we fallback to sniffing the CSV dialect (DEBUG/WARN/INFO), if it's any more error-prone.